lib/jobs/sftp-job.js

```diff
@@ -0,0 +1,119 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+module.exports = sftpJobFactory;
+
+var Client = require('ssh2').Client;
+di.annotate(sftpJobFactory, new di.Provide('Job.Sftp'));
+di.annotate(sftpJobFactory, new di.Inject(
+    'Job.Base',
+    'Util',
+    'Logger',
+    'Assert',
+    'Promise',
+    'Services.Encryption',
+    'Services.Waterline'
+));
+
+function sftpJobFactory(
+    BaseJob,
+    util,
+    Logger,
+    assert,
+    Promise,
+    cryptService,
+    waterline
+) {
+    var logger = Logger.initialize(sftpJobFactory);
+
+    function SftpJob(options, context, taskId) {
+        SftpJob.super_.call(this, logger, options, context, taskId);
+        assert.string(options.fileSource);
+        assert.string(options.fileDestination);
+
+        assert.string(this.context.target);
+        this.nodeId = this.context.target;
+        this.timeout = options.timeout;
+        this.fileSource = options.fileSource;
+        this.fileDestination = options.fileDestination;
+    }
+    util.inherits(SftpJob, BaseJob);
+
+    SftpJob.prototype._run = function run() {
+        var self = this;
+        return waterline.nodes.needByIdentifier(this.context.target)
+        .then(function(node) {
+                return self.runSftp(node.sshSettings, new Client());
+            })
+            .then(function() {
+                self._done();
+            })
+            .catch(function(e) {
+                self._done(e);
+            });
+    };
+
+    SftpJob.prototype.runSftp = function(sshSettings, sshClient) {
+        var self = this;
+        return new Promise(function(resolve, reject) {
+            if(self.timeout) {
```

Comment:
OK, thanks. Then it will be better to move the default timeout to RackHD code, so other user doesn't need to search to `ssh` library to check its default behavior.